docs/sphinx/source/conf.py

```diff
@@ -445,8 +445,9 @@ def make_github_url(file_name):
         if file_name.split("/")[-1] == "index":
             example_file = "README.rst"
         else:
+            example_folder = file_name.split("/")[-2]
             example_file = file_name.split("/")[-1].replace('.rst', '.py')
-        target_url = URL_BASE + "docs/examples/" + example_file
+        target_url = URL_BASE + "docs/examples/" + example_folder + "/" + example_file  # noqa: E501
```

Comment:
One issue: `example_folder` is only defined in the `else` branch here.

Another issue is that the conditional might need to check `file_name.split("/")[-1] == "index.rst"`?  Not 100% sure, but worth trying :)